pkg/jobrunaggregator/gcs_bucket.go

```diff
@@ -0,0 +1,130 @@
+package jobrunaggregator
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	prowjobv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	utiltrace "k8s.io/utils/trace"
+)
+
+type JobRun struct {
+	ProwJob   *prowjobv1.ProwJob
+	GCSBucket string
+}
+
+func (o *JobRunAggregatorOptions) ReadProwJob(ctx context.Context) ([]JobRun, error) {
+	fmt.Printf("Reading prowjobs for job %v.\n", o.JobName)
+
+	prowJobPaths, err := o.getProwJobPathsForJob(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	jobRuns := []JobRun{}
+	bkt := o.GCSClient.Bucket(openshiftCIBucket)
+	// Iterate through the ProwJob paths, retrieve the objects and decode them into a struct for further processing
+	for _, prowJobPath := range prowJobPaths {
+		prowJob, err := getProwJob(ctx, bkt, prowJobPath)
+		if err != nil {
+			return nil, err
+		}
+
+		logrus.Infof("Decoded %s into struct", prowJobPath)
+
+		jobRuns = append(jobRuns, JobRun{
+			ProwJob:   prowJob,
+			GCSBucket: prowJobPath,
+		})
+	}
+
+	return jobRuns, nil
+}
+
+func (o *JobRunAggregatorOptions) traceFields() []utiltrace.Field {
+	return []utiltrace.Field{
+		{Key: "jobName", Value: o.JobName},
+	}
+}
+
+func (o *JobRunAggregatorOptions) getProwJobPathsForJob(ctx context.Context) ([]string, error) {
+	trace := utiltrace.New("GetProwJobs", o.traceFields()...)
+	defer trace.LogIfLong(500 * time.Millisecond)
+
+	prowJobJsonFiles := []string{}
+
+	bkt := o.GCSClient.Bucket(openshiftCIBucket)
+
+	query := &storage.Query{
+		// This ends up being the equivalent of:
+		// https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-nightly-4.9-upgrade-from-stable-4.8-e2e-metal-ipi-upgrade
```

Comment:
as noted in the past if you want to support e.g. presubmits it's best to load the prow config and use the prow libraries to determine this path